backend/owa.ts

```diff
@@ -1,8 +1,21 @@
 import { session as Session, BrowserWindow } from "electron";
 
+const kCookieName = "X-OWA-CANARY";
+const kHotmailServer = "outlook.live.com";
+const reOWAPath = /^\/owa\/(\d+\/){0,1}/;
+
 export async function fetchSessionData(partition: string, url: string, interactive: boolean) {
   let session = Session.fromPartition(partition);
-  let response = await session.fetch(url + 'sessiondata.ashx', { method: 'POST' });
+  let urlObj = new URL(url);
+  let cookies = await session.cookies.get({ name: 'X-OWA-CANARY' });
+  if (cookies.length) {
```

Comment:
@benbucksch Do you think we should check if there are existing cookies and setting the URL based on the existing cookies? Because that's the last issue I have with this. Because

https://github.com/mustang-im/mustang/blob/2dca9083b3d6b0c91d18263404799a821731674e/app/logic/Mail/AutoConfig/saveConfig.ts#L15

Calls it with interactive true and was going to use that to determine whether or not to set the URL based on the domain and path of the existing cookies.

And that causes an issue where when you click `Next` it shows popup again because on the first fetch there's an error and interactive is set to true but it is still saving.

Reply:
Sorry, I don't understand any part of the question.